packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx

```diff
@@ -0,0 +1,191 @@
+    leftPosition
+  );
+  const rightCartographic = globe.ellipsoid.cartesianToCartographic(
+    rightPosition
+  );
+
+  geodesic.setEndPoints(leftCartographic, rightCartographic);
+  const pixelDistance = geodesic.surfaceDistance;
+
+  // Find the first distance that makes the scale bar less than 100 pixels.
+  const maxBarWidth = 100;
+  let distance;
+  for (let i = distances.length - 1; !isNumber(distance) && i >= 0; --i) {
+    if (distances[i] / pixelDistance < maxBarWidth) {
+      distance = distances[i];
+    }
+  }
+
+  if (isNumber(distance)) {
+    let label = '';
+    if (distance >= 1000) {
+      label = `${(distance / 1000).toString()} ${kiloMetersUnit}`;
+    } else {
+      label = `${distance.toString()} ${metersUnit}`;
+    }
+
+    scale.barWidth = (distance / pixelDistance) | 0;
+    scale.distanceLabel = label;
+  }
+
+  setScaleData(scale);
+};
+/* eslint-enable @typescript-eslint/no-magic-numbers */
+
+export const ScaleTrackerTool: React.FC<RScaleTrackerToolProps> = (props) => {
+  const mapViewer: Viewer = useMap();
+  const [scaleData, setScaleData] = useState<IScaleData>({
+    barWidth: undefined,
+    distanceLabel: undefined,
+    lastLegendUpdate: -1,
+  });
+
+  useEffect(() => {
+    const setFromEvent = (e: MouseEvent): void => {
+      updateDistanceLegendCesium(
+        mapViewer,
+        scaleData,
+        setScaleData,
+        props.locale
+      );
+    };
+
+    const helper = new EventHelper();
+    const tileLoadHandler = (event: number): void => {
+      // console.log("Tiles to load: " + event, mapViewer.scene.globe.tilesLoaded);
+      if (mapViewer.scene.globe.tilesLoaded) {
+        setFromEvent(new MouseEvent('mouse'));
+        helper.removeAll();
+      }
+    };
+    // Register tiles loader handler for initial load because globe.pick returning undefined
+    // see here https://community.cesium.com/t/globe-pick-returning-undefined/6616
+    helper.add(mapViewer.scene.globe.tileLoadProgressEvent, tileLoadHandler);
+
+    mapViewer.camera.moveEnd.addEventListener(setFromEvent);
+    // mapViewer.camera.changed.addEventListener(setFromEvent);
+
+    return (): void => {
+      try {
+        mapViewer.camera.moveEnd.removeEventListener(setFromEvent);
+        // mapViewer.camera.changed.removeEventListener(setFromEvent);
+      } catch (e) {
+        console.log('CESIUM camera "moveEnd" remove listener failed', e);
+      }
+    };
+  }, [mapViewer, props.locale, scaleData]);
+
+  return (
+    <div className="scalePosition">
+      {isNumber(scaleData.barWidth) && (
+        <>
+          <div className="distance-legend-label">{scaleData.distanceLabel}</div>
+          <div
+            className="distance-legend-scale-bar"
+            style={{
+              height: '2px',
+              width: `${scaleData.barWidth.toString()}px`,
+              // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+              left: `${(5 + (125 - scaleData.barWidth) / 2).toString()}px`,
```

Comment:
I think we should extract the calculation from the jsx and let it sit in its own function

Reply:
Yep, U r right
Done